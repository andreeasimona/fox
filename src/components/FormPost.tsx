import React from "react";
import { IFormPost } from "types/post";
import { useHistory } from "react-router-dom";

const FormPost = (props: IFormPost) => {
  const {
    post: { title, content, lat, long, image_url },
    onChange,
    onSubmit,
    disabled
  } = props;
  const history = useHistory();
  return (
    <form>
      <div className="form-group">
        <label>Title</label>
        <input
          className="form-control"
          value={title}
          name="title"
          onChange={onChange}
        />
      </div>

      <div className="form-group">
        <label>Content</label>
        <input
          className="form-control"
          value={content}
          name="content"
          onChange={onChange}
        />
      </div>

      <div className="form-group">
        <label>Latitude</label>
        <input
          className="form-control"
          value={lat}
          name="lat"
          onChange={onChange}
        />
      </div>

      <div className="form-group">
        <label>Longitude</label>
        <input
          className="form-control"
          value={long}
          name="long"
          onChange={onChange}
        />
      </div>

      <div className="form-group">
        <label>Image url</label>
        <input
          className="form-control"
          value={image_url}
          name="image_url"
          onChange={onChange}
        />
      </div>
      <hr />
      <div className="container">
        <div className="row">
          <div className="col-11">
            <button
              type="button"
              className="btn btn-secondary"
              onClick={() => history.goBack()}
            >
              Back
            </button>
          </div>
          <div className="col-1">
            <button
              type="button"
              className="btn btn-primary"
              onClick={onSubmit}
              disabled={disabled}
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </form>
  );
};

export default FormPost;
