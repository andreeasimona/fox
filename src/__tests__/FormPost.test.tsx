import React from "react";
import { MemoryRouter } from "react-router-dom";
import FormPost from "../components/FormPost";
import { render } from "@testing-library/react";

describe("FormPost", () => {
  it("renders the form post without values and submit disabled", () => {
    const mockProps = {
      post: {},
      disabled: true,
      onChange: () => jest.fn(),
      onSubmit: () => jest.fn()
    };
    const wrapper = render(
      <MemoryRouter>
        <FormPost {...mockProps} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders the form post with values", () => {
    const mockProps = {
      post: {
        title: "title",
        content: "content",
        long: "long",
        lat: "lat",
        image_url: "image_url"
      },
      disabled: false,
      onChange: () => jest.fn(),
      onSubmit: () => jest.fn()
    };
    const wrapper = render(
      <MemoryRouter>
        <FormPost {...mockProps} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
