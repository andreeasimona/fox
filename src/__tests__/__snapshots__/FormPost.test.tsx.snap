// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPost renders the form post with values 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="form-group"
        >
          <label>
            Title
          </label>
          <input
            class="form-control"
            name="title"
            value="title"
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Content
          </label>
          <input
            class="form-control"
            name="content"
            value="content"
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Latitude
          </label>
          <input
            class="form-control"
            name="lat"
            value="lat"
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Longitude
          </label>
          <input
            class="form-control"
            name="long"
            value="long"
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Image url
          </label>
          <input
            class="form-control"
            name="image_url"
            value="image_url"
          />
        </div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-11"
            >
              <button
                class="btn btn-secondary"
                type="button"
              >
                Back
              </button>
            </div>
            <div
              class="col-1"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="form-group"
      >
        <label>
          Title
        </label>
        <input
          class="form-control"
          name="title"
          value="title"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Content
        </label>
        <input
          class="form-control"
          name="content"
          value="content"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Latitude
        </label>
        <input
          class="form-control"
          name="lat"
          value="lat"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Longitude
        </label>
        <input
          class="form-control"
          name="long"
          value="long"
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Image url
        </label>
        <input
          class="form-control"
          name="image_url"
          value="image_url"
        />
      </div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-11"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Back
            </button>
          </div>
          <div
            class="col-1"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FormPost renders the form post without values and submit disabled 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="form-group"
        >
          <label>
            Title
          </label>
          <input
            class="form-control"
            name="title"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Content
          </label>
          <input
            class="form-control"
            name="content"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Latitude
          </label>
          <input
            class="form-control"
            name="lat"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Longitude
          </label>
          <input
            class="form-control"
            name="long"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label>
            Image url
          </label>
          <input
            class="form-control"
            name="image_url"
            value=""
          />
        </div>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-11"
            >
              <button
                class="btn btn-secondary"
                type="button"
              >
                Back
              </button>
            </div>
            <div
              class="col-1"
            >
              <button
                class="btn btn-primary"
                disabled=""
                type="button"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="form-group"
      >
        <label>
          Title
        </label>
        <input
          class="form-control"
          name="title"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Content
        </label>
        <input
          class="form-control"
          name="content"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Latitude
        </label>
        <input
          class="form-control"
          name="lat"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Longitude
        </label>
        <input
          class="form-control"
          name="long"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Image url
        </label>
        <input
          class="form-control"
          name="image_url"
          value=""
        />
      </div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-11"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Back
            </button>
          </div>
          <div
            class="col-1"
          >
            <button
              class="btn btn-primary"
              disabled=""
              type="button"
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
